### PR TITLE
Add an additional abstract test.

### PR DESCRIPTION
**Pull request by floitsch**

### CODE REVIEW

File: tests/abstract4-test.toit [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+class C extends B:
+  foo x=null:
+    return super
+
+class D:
+  foo:
+    return 33
+
+confuse x: return x
```
Comment (kasperl) [inline tests/abstract4-test.toit:21]: I think we have a better way of confusing in `tests/toitp/confuse.toit`. Maybe we should use that in more places?

File: tests/abstract4-test.toit [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+main:
+  c := confuse C
+  expect_equals 1 c.foo
+  expect_equals 1 (c.foo 1)
+
+  d := confuse D
+  expect_equals 33 d.foo
```
Comment (kasperl) [inline tests/abstract4-test.toit:29]: What does this test? There is nothing abstract about foo in this case?

**Review by kasperl — APPROVED** [review]
Looks good to me.

File: tests/abstract4-test.toit (hunk 1 above)
Comment (floitsch) [reply tests/abstract4-test.toit:21]: I think we should eventually go through all `confuse` methods and make it use something similar.
I'm not too keen on using something from the `toitp` directory.
In another PR...

File: tests/abstract4-test.toit (hunk 2 above)
Comment (floitsch) [reply tests/abstract4-test.toit:29]: Made `D` extend `C`.
done.

**Pull request merged**